multiplatform-movies/common/src/commonMain/kotlin/com.novoda.movies.core/NetworkingDependencyProvider.kt

```diff
@@ -0,0 +1,27 @@
+package com.novoda.movies.core
+
+import io.ktor.client.HttpClient
+import io.ktor.client.HttpClientConfig
+import io.ktor.client.features.defaultRequest
+import io.ktor.client.request.parameter
+
+private const val KEY_API_SECRET: String = "api_key"
+private const val API_SECRET: String = "your themoviedb api secret" //https://developers.themoviedb.org/3
+private const val KEY_AUTH_INTERCEPTOR = "auth_interceptor"
+
+class NetworkingDependencyProvider {
+
+    fun provideAuthenticatedClient() = HttpClient {
+        authenticatedRequestsEngine()
+    }
+
+    private fun HttpClientConfig<*>.authenticatedRequestsEngine() {
+        engine {
+            install(KEY_AUTH_INTERCEPTOR) {
```

Comment:
Using this api we can easily share the configuration for the ktor http client, in this case we want to add the `api_key` to each request.

Reply:
Can this not come from the build process?

Reply:
Yes! Next step is making out build properties plugin multi platform aware : 💯